docs/metrics/README.md

```diff
@@ -0,0 +1,61 @@
+# OpenTelemetry .NET Metrics
+
+## Best Practices
+
+- When emitting metrics with dimensions, ALWAYS provide the dimensions in an
+  ascending sorted order (order by the dimension key) for best performance.
+
+```csharp
+// Sorted in an ascending order by dimension key: `color` comes before `name`
+MyFruitCounter.Add(1, new("color", "red"), new("name", "apple")); // <--- DO THIS
+
+MyFruitCounter.Add(1, new("name", "apple"), new("color", "red")); // <--- DON'T DO THIS
+```
+
+- When emitting metrics with more than three dimensions, use `TagList` for best
+  performance.
+
+```csharp
+var tags = new TagList
```

Comment:
Also hint that, SDK offers highest performance for up to 3 tags, high performance for up to 8 tags, and for tags > 8, there is heap-allocation in the hot path.

Add a tip that "if you are exceeding 8 tags for a metric, check if some of the attributes are better modelled as Resources - machine name, environment etc. Those should be modelled as Resource (and link to Resource doc).".